### PR TITLE
Return total scaled volume in SAL metric

### LINKED ISSUE #344: Structure parameter in SAL verification metric
When calculating scaled volume (V(R)), it should be proportional to the second moment of the precipitation field, not the first.

### PR DESCRIPTION
**Pull request by EsmailGhaemi**
Fixed #344 (S parameter edited, L parameter modified for dealing with nan values)

Now the scaled volume function directly returns total scaled volume (instead of scaled voulme for each object). Additionally, for the weighted distance function, np.nansum() is used to avoid the effect of nan values.

### CODE REVIEW

File: pysteps/verification/salscores.py [hunk 1]
```diff
@@ -155,11 +155,11 @@ def sal_structure(
     observation_objects = _sal_detect_objects(
         observation, thr_factor, thr_quantile, tstorm_kwargs
     )
-    prediction_volume = _sal_scaled_volume(prediction_objects).sum()
-    observation_volume = _sal_scaled_volume(observation_objects).sum()
+    prediction_volume = sal_scaled_volume(prediction_objects)
```
Comment (dnerini) [inline pysteps/verification/salscores.py]: why did you rename the calls to `_sal_scaled_volume`? this is causing the NameErrors in the tests, can you revert this change?

File: pysteps/verification/salscores.py [hunk 2]
```diff
@@ -155,11 +155,11 @@ def sal_structure(
...
+    observation_volume = sal_scaled_volume(observation_objects)
     nom = prediction_volume - observation_volume
     denom = prediction_volume + observation_volume
-    return np.divide(nom, (0.5 * denom))
+    return nom / (0.5 * denom)
```
Comment (dnerini) [inline pysteps/verification/salscores.py]: I changed this to `np.divide()` in #327 because we want this function to raise an error instead of returning `Inf`, any particular reason to change it back?

**Review by dnerini — CHANGES_REQUESTED** [review]
Hi @EsmailGhaemi! Very nice seeing you back here :-) Thanks a lot for the fix! I have few questions on some changes you introduced.

Tests are currently failing because of NameErrors (see my comment below). Let me know if you need any help on this.

File: pysteps/verification/salscores.py (hunk 2 above)
Comment (EsmailGhaemi) [reply pysteps/verification/salscores.py]: Hi @dnerini, thank you.
You are right. I've applied black and edited these lines.
Merry Christmas :)

**Comment by dnerini** [comment]
Thanks @EsmailGhaemi for the changes. Could you just spend few more words on what you did? It's not obvious to me what you changed, why, and what kind of impact we should expect. This will help to describe the changes in the release notes.

**Comment by EsmailGhaemi** [comment]
Yes, of course.
In the previous version, the scaled volume was calculated for each object separately, and then in sal_srtucture function we summed all of them. However, when calculating total volume (V_R in equation 8, Wernli et al., 2008), we should multiply each object's scaled volume (v_n) to the amount of precipitation for that object (R_n) as well.
Now the scaled volume function directly returns total scaled volume (instead of scaled voulme for each object). Additionally, for the weighted distance function, np.nansum() is used to avoid the effect of nan values.

**Comment by dnerini** [comment]
Brilliant, thanks, I've update the description above.

**Review by dnerini — APPROVED** [review]
